pkg/test/framework/components/apps/kube.go

```diff
@@ -661,14 +660,24 @@ func (a *kubeApp) Call(e AppEndpoint, opts AppCallOptions) ([]*echo.ParsedRespon
 		},
 	})
 	if err != nil {
-		return nil, err
+		// Return resp instead of nil when err != nil. Needed because when a TCP request is rejected
+		// due to authorization, the response body will be empty. This doesn't mean that the request is
+		// not successful. For example, if the authorization policy is to reject this request, this is
+		// considered successful.
+		// Callers other than RBAC-related callers should already check for resp != nil first.
+		return resp, err
 	}
 
 	if len(resp) != 1 {
 		return nil, fmt.Errorf("unexpected number of responses: %d", len(resp))
 	}
+
 	if !resp[0].IsOK() {
-		return nil, fmt.Errorf("unexpected response status code: %s", resp[0].Code)
+		// Return resp instead of nil when the response code is not 200. This is because in some cases,
+		// such as when a request is expected to be rejected due to authorization, the response code is
+		// 403, and this is expected/considered successful.
+		// Callers other than RBAC-related callers should already check for resp != nil first.
+		return resp, fmt.Errorf("unexpected response status code: %s", resp[0].Code)
```

Comment:
We should move the error checking into call or fail and make the call function just return the resp without error checking. I do the same thing in a PR I have out

Reply:
Could you link your PR?

Reply:
https://github.com/istio/istio/pull/13099/files

Reply:
I see what you mean. Yea, I think your approach is better. I dropped a comment on your PR

Reply:
Let me know what you guys think @nmittler @ozevren